src/main/java/com/teragrep/pth10/steps/teragrep/TeragrepBloomStep.java

```diff
@@ -161,14 +162,14 @@ private Dataset<Row> estimateSize(Dataset<Row> dataset) {
                 .agg(functions.approxCountDistinct("token").as(outputCol));
     }
 
-    public Dataset<Row> aggregate(Dataset<Row> dataset) {
-
-        FilterTypes filterTypes = new FilterTypes(this.zeppelinConfig);
-
-        BloomFilterAggregator agg = new BloomFilterAggregator(inputCol, estimateCol, filterTypes.sortedMap());
-
-        return dataset.groupBy("partition").agg(agg.toColumn().as("bloomfilter"));
-
+    public Dataset<Row> aggregate(final Dataset<Row> dataset) {
+        final ColumnBinaryListingDataset inputColumnToBytes = new ColumnBinaryListingDataset(dataset, inputCol);
+        final BloomFilterAggregator agg = new BloomFilterAggregator(
```

Comment:
do not use method name as a variable name GroupedDataset has .agg as a method name and it is used on line 172

Reply:
renamed to variable to bloomFilterAggregator